evm-contracts/src/v0.6/dev/BlockhashStore.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity 0.6.2;
+
+/**
+ * @title BlockhashStore
+ * @notice This contract provides a way to access blockhashes older than
+ * the 256 block limit imposed by the BLOCKHASH opcode.
+ * You may assume that any blockhash stored by the contract is correct.
+ */
+contract BlockhashStore {
+
+  mapping(uint => bytes32) internal s_blockhashes;
+
+  /**
+   * @notice stores blockhash of a given block, assuming it is available through BLOCKHASH
+   * @param n the number of the block whose blockhash should be stored
+   */
+  function store(uint256 n) public {
+    bytes32 h = blockhash(n);
+    require(h != 0x0, "blockhash(n) failed");
+    s_blockhashes[n] = h;
+  }
+
+
+  /**
+   * @notice stores blockhash of the earliest block still available through BLOCKHASH.
+   */
+  function storeEarliest() external {
+    store(block.number - 256);
+  }
+
+  /**
+   * @notice stores blockhash after verifying blockheader of child/subsequent block
+   * @param n the number of the block whose blockhash should be stored
+   * @param header the rlp-encoded blockheader of block n+1. We verify its correctness by checking
```

Comment:
We probably don't care, but in my hands, linebreaks in `@decorated` docstrings like this result in truncated annotations in the resulting html docs. I think `that` should be indented so it's clear the line belongs to the preceding `@param`, though.